docs/root/configuration/access_log.rst

```diff
@@ -17,16 +17,93 @@ Access logs are configured as part of the :ref:`HTTP connection manager config
+:ref:`"format dictionaries" <config_access_log_format_dictionaries>`. In both cases, the command operators
+are used to extract the relevant data, which is then inserted into the specified log format.
+Only one access log format may be specified at a time.
+
+.. _config_access_log_format_strings:
+
+Format Strings
+--------------
+
+Format strings are plain strings, specified using the ``format`` key. They may contain
+either command operators or other characters interpreted as a plain string. 
+The access log formatter does not make any assumptions about a new line separator, so one
 has to specified as part of the format string.
 See the :ref:`default format <config_access_log_default_format>` for an example.
-Note that the access log line will contain a '-' character for every not set/empty value.
 
-The same format strings are used by different types of access logs (such as HTTP and TCP). Some
+.. _config_access_log_default_format:
+
+Default Format String
+---------------------
+
+If custom format string is not specified, Envoy uses the following default format:
+
+.. code-block:: none
+
+  [%START_TIME%] "%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%"
+  %RESPONSE_CODE% %RESPONSE_FLAGS% %BYTES_RECEIVED% %BYTES_SENT% %DURATION%
+  %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)% "%REQ(X-FORWARDED-FOR)%" "%REQ(USER-AGENT)%"
+  "%REQ(X-REQUEST-ID)%" "%REQ(:AUTHORITY)%" "%UPSTREAM_HOST%"\n
+
+Example of the default Envoy access log format:
+
+.. code-block:: none
+
+  [2016-04-15T20:17:00.310Z] "POST /api/v1/locations HTTP/2" 204 - 154 0 226 100 "10.0.35.28"
+  "nsq2http" "cc21d9b0-cf5c-432b-8c7e-98aeb7988cd2" "locations" "tcp://10.0.2.1:80"
+
+.. _config_access_log_format_dictionaries:
+
+Format Dictionaries
+-------------------
+
+Format dictionaries are dictionaries that specify a structured access log output format,
+specified using the ``json_format`` key. This allows logs to be output in a structured format
+such as JSON.
+Similar to format strings, command operators are evaluated and their values inserted into the format
+dictionary to construct the log output.
+
+For example, with the following format provided in the configuration:
+
+.. code-block:: json
+
+  {
+    "config": {
+      "json_format": {
+          "protocol": "%PROTOCOL%",
+          "duration": "%DURATION%",
+          "my_custom_header": "%REQ(MY_CUSTOM_HEADER)%"
+      }
+    }
+  }
+  
+The following JSON object would be written to the log file:
+
+.. code-block:: json
+
+  {"protocol": "HTTP/1.1", "duration": "123", "my_custom_header": "value_of_MY_CUSTOM_HEADER"}
+
+This allows you to specify a custom key for each command operator.
+
+Format dictionaries have the following restrictions:
+
+* The dictionary must map strings to strings (specifically, strings to command operators). Nesting is not currently supported.
+* Each value in the dictionary can only contain one command operator.
+
+
+Command Operators
```

Comment:
FWIW, I think "command operator" is not the right terminology, but I see it was existing before this PR. I probably would have used "format expressions" or the like.

Reply:
I'll see if it's pretty straightforward to change this. I personally prefer "format specifiers" since it lines up with the C terminology

Reply:
I left this as is for now

Reply:
Sure, this was more just a non-actionable observation.